spec/controllers/items_controller_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'rails_helper'
+
+RSpec.describe ItemsController, type: :controller do
+  describe '#index' do
+    let!(:items) { create_list(:item, 7) }
+    context 'home' do
+      before { get :index }
+
+      it { expect(controller.items).to contain_exactly(*items[0..5]) }
+
+      it { expect(response).to render_template(:index) }
+    end
+
+    context 'search' do
+      let(:item) { create(:item) }
+      before { get :index, params: { q: { name_cont: item.name } } }
+
+      it { expect(controller.items).to contain_exactly(item) }
+    end
+  end
+
+  describe '#show' do
+    let(:item) { create(:item) }
+    let(:call_request) { get :show, params: { id: item.id } }
+
+    it_behaves_like 'an action rendering view'
+  end
+
+  describe '#destroy' do
+    let!(:item) { create(:item) }
+
+    context 'when logged in as admin' do
+      let!(:admin) { create(:admin) }
+      before do
+        sign_in admin
+        delete :destroy, params: { id: item.id }
+      end
+
+      it { expect(controller.items).not_to include(item) }
```

Comment:
consider selleo controller tests to refactor this
https://github.com/Selleo/selleo-controller_tests#an-action-destroying-object